packages/terra-pills/src/PillList.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  KEY_SPACE,
+  KEY_RETURN,
+} from 'keycode-js';
+import PropTypes from 'prop-types';
+import classNamesBind from 'classnames/bind';
+import ThemeContext from 'terra-theme-context';
+import styles from './Pill.module.scss';
+import Pill from './Pill';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+   * The aria label string value for the pills container.
+   */
+  ariaLabel: PropTypes.string.isRequired,
+  /**
+   * The content to be shown in the pills container. Should only contain the sub-component pills
+   */
+  children: PropTypes.node,
+  /**
+   * Determines if the Pill List rolls up or not.
+   */
+  isCollapsible: PropTypes.bool,
+};
+
+const defaultProps = {
+  isCollapsible: false,
+};
+
+const PillList = (props) => {
+  const {
+    ariaLabel,
+    children,
+    isCollapsible,
+  } = props;
+
+  const theme = React.useContext(ThemeContext);
+  const pillListRef = useRef();
+  const rollUpPillRef = useRef();
+  const [rollUpCount, setRollUpCount] = useState(0);
+  const [isRollUpRequired, setIsRollUpRequired] = useState(true);
+
+  const handleCollapse = () => {
+    let pillListHeight = pillListRef.current.getBoundingClientRect().height;
```

Comment:
Why is the rollup logic based on height?

Reply:
I tried a couple of different approaches but i felt the most appropriate for the scenario where the PillList is rollable and the maximum number of pills including the roll up pill should be incorporated only in the first line.
I am definitely open to suggestion for other alternative approaches.